eng/bomgenerator/src/main/java/com/azure/tools/bomgenerator/Utils.java

```diff
@@ -183,6 +189,46 @@ static BomDependencyNoVersion toBomDependencyNoVersion(BomDependency bomDependen
     }
 
     static List<BomDependency> parsePomFileContent(Reader responseStream) {
+        List<BomDependency> bomDependencies = new ArrayList<>();
+
+        ObjectMapper mapper = new XmlMapper();
+        try {
+            HashMap<String, Object> value = mapper.readValue(responseStream, HashMap.class);
+            Object packagingProp = value.getOrDefault("packaging", null);
+            if(packagingProp != null && packagingProp.toString().equalsIgnoreCase("pom")) {
+                return parsePomFileContent(responseStream);
+            }
+
+            // else we have a traditional POM file so we are responsible for parsing its contents.
+            HashMap<String, Object> dependenciesTag = (HashMap<String, Object>)value.getOrDefault("dependencies", null);
+            if(dependenciesTag == null) {
+                return null;
+            }
+
+            ArrayList<HashMap<String, Object>> dependencies =  (ArrayList<HashMap<String, Object>>) dependenciesTag.getOrDefault("dependency", null);
+            for(HashMap<String, Object> dependency: dependencies) {
+                String groupId = (String) dependency.getOrDefault("groupId", null);
+                String artifactId = (String) dependency.getOrDefault("artifactId", null);
+                String version = (String) dependency.getOrDefault("version", null);
+                String scope = (String) dependency.getOrDefault("scope", ScopeType.COMPILE.toString());
+                ScopeType scopeType = ScopeType.COMPILE;
+                switch(scope) {
+                    case "test" : scopeType = ScopeType.TEST;
+                    break;
+
+                    default: scopeType = ScopeType.COMPILE;
```

Comment:
Missing a `break`

Reply:
Also, do we not care about other types of scope such as `provided`?

Reply:
As of now, no. We may eventually need to support both import and provided but currently we have no use case for them.